compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java

```diff
@@ -490,15 +490,19 @@ private Package createNewPackage() {
                 CompilerContext compilerContext = project.projectEnvironmentContext()
                         .getService(CompilerContext.class);
                 PackageCache packageCache = PackageCache.getInstance(compilerContext);
+
+                io.ballerina.projects.environment.PackageCache service =
+                        project.projectEnvironmentContext().environment().getService(
+                        io.ballerina.projects.environment.PackageCache.class);
+
                 for (ResolvedPackageDependency dependency : diff) {
                     for (ModuleId moduleId : dependency.packageInstance().moduleIds()) {
                         if (!dependency.packageInstance().descriptor().isLangLibPackage()) {
                             Module module = dependency.packageInstance().module(moduleId);
                             PackageID packageID = module.descriptor().moduleCompilationId();
                             // remove the module from the compiler packageCache
                             packageCache.remove(packageID);
-                            // we need to also reset the module in the project environment packageCache
```

Comment:
Let's keep this comment as it explains what we are doing

Reply:
Added the comment back